src/driver/mpas_subdriver.F

```diff
@@ -109,38 +111,66 @@ end subroutine xml_stream_get_attributes
       readNamelistArg = .false.
       readStreamsArg = .false.
 
-      nArgs = command_argument_count()
-      iArg = 1
-      do while (iArg < nArgs)
-         call get_command_argument(iArg, argument)
-         if (len_trim(argument) == 0) exit
-
-         if ( trim(argument) == '-n' ) then
-            iArg = iArg + 1
+      ! If provided, error check the namelistFileParam and copy it to namelistFile to override default
+      if (present(namelistFileParam)) then
+         if (len_trim(namelistFileParam) <= 0) then
+            write (0,*) 'WARNING: mpas_init argument namelistFileParam has 0 length and will be ignored'
+         else if (len_trim(namelistFileParam) > len(namelistFile)) then
+            write (0,*) 'CRITICAL ERROR: mpas_init argument namelistFileParam is too long for namelistFile variable.'
```

Comment:
Here and for the streamsFile handling, we could print the length of the argument as well as the maximum acceptable length (i.e., the length of the local variable to which the argument will be assigned). This could help reduce debugging time for cases, e.g., where a long string with corrupted contents or other garbage is accidentally passed to the function: being told that the namelist filename is, say, 1024 characters long could lead to a quicker diagnosis of the issue.

Reply:
@mgduda, how does it look now after I pushed? I pretty much wrote what I would put in a debug print statement for my use, so there's probably a better format.

Reply:
Since the message is meant for human consumption, I've offered an alternative in a review comment that (in my opinion) may state the issue a little more clearly.